batch2/batch/driver/k8s.py

```diff
@@ -1,112 +1,24 @@
-        self._read_pod_log = self._wrap_k8s(k8s_api.read_namespaced_pod_log)
-        self._read_pod_status = self._wrap_k8s(k8s_api.read_namespaced_pod_status)
-        self._list_pods = self._wrap_k8s(k8s_api.list_namespaced_pod)
-        self._list_pvcs = self._wrap_k8s(k8s_api.list_namespaced_persistent_volume_claim)
-        self._get_pod = self._wrap_k8s(k8s_api.read_namespaced_pod)
-        self._get_pvc = self._wrap_k8s(k8s_api.read_namespaced_persistent_volume_claim)
-        self._read_secret = self._wrap_k8s(k8s_api.read_namespaced_secret,
-                                           pc.Summary('batch_read_secret',
-                                                      'Batch k8s read secret latency in seconds'))
-
-    async def delete_pod(self, name):
-        assert name is not None
-        return await self._delete_pod(name=name)
-
-    async def delete_pvc(self, name):
-        assert name is not None
-        return await self._delete_pvc(name=name)
-
-    async def create_pod(self, *args, **kwargs):
-        return await self._create_pod(*args, **kwargs)
-
-    async def create_pvc(self, *args, **kwargs):
-        return await self._create_pvc(*args, **kwargs)
-
-    async def read_pod_log(self, *args, **kwargs):
-        return await self._read_pod_log(*args, **kwargs)
-
-    async def read_pod_status(self, *args, **kwargs):
-        return await self._read_pod_status(*args, **kwargs)
-
-    async def list_pods(self, *args, **kwargs):
-        return await self._list_pods(*args, **kwargs)
-
-    async def list_pvcs(self, *args, **kwargs):
-        return await self._list_pvcs(*args, **kwargs)
-
-    async def get_pod(self, *args, **kwargs):
-        return await self._get_pod(*args, **kwargs)
-
-    async def get_pvc(self, *args, **kwargs):
-        return await self._get_pvc(*args, **kwargs)
+        self.k8s_client = k8s_client
 
     async def read_secret(self, *args, **kwargs):
-        return await self._read_secret(*args, **kwargs)
-
-    def _wrap_k8s(self, fun, pc_summary=NoSummary()):
-        async def wrapped(*args, **kwargs):
+        if '_request_timeout' not in kwargs:
+            kwargs['_request_timeout'] = self.timeout
+        if 'namespace' not in kwargs:
+            kwargs['namespace'] = self.namespace
+        with READ_SECRET_TIME.time():
             try:
-                if '_request_timeout' not in kwargs:
-                    kwargs['_request_timeout'] = self.timeout
-                if 'namespace' not in kwargs:
-                    kwargs['namespace'] = self.namespace
-                with pc_summary.time():
-                    return (await blocking_to_async(self.blocking_pool,
-                                                    fun,
-                                                    *args,
-                                                    **kwargs),
-                            None)
+                return (self.k8s_client.read_namespaced_secret(*args, **kwargs),
```

Comment:
Shouldn't this have an await somewhere?

Reply:
Indeed, fixed.  I wish pylint could check for this, I make this mistake constantly.